Utils/azuretools-core/src/com/microsoft/azuretools/authmanage/AuthMethodManager.java

```diff
@@ -200,35 +224,56 @@ public AuthMethodDetails getAuthMethodDetails() {
         return this.authMethodDetails;
     }
 
-    public void setAuthMethodDetails(AuthMethodDetails authMethodDetails) throws IOException {
+    public synchronized void setAuthMethodDetails(AuthMethodDetails authMethodDetails) throws IOException {
         cleanAll();
         this.authMethodDetails = authMethodDetails;
         saveSettings();
         //if (isSignedIn()) notifySignInEventListener();
     }
 
-    private void loadSettings() {
+    private AuthMethodDetails loadSettings() {
         System.out.println("loading authMethodDetails...");
         try {
             FileStorage fs = new FileStorage(FILE_NAME_AUTH_METHOD_DETAILS, CommonSettings.getSettingsBaseDir());
             byte[] data = fs.read();
             String json = new String(data);
             if (json.isEmpty()) {
                 System.out.println(FILE_NAME_AUTH_METHOD_DETAILS + " is empty");
-                authMethodDetails = new AuthMethodDetails();
-                return;
+                return new AuthMethodDetails();
             }
-            authMethodDetails = JsonHelper.deserialize(AuthMethodDetails.class, json);
+
+            return JsonHelper.deserialize(AuthMethodDetails.class, json);
         } catch (IOException ignored) {
             System.out.println("Failed to loading authMethodDetails settings. Use defaults.");
-            authMethodDetails = new AuthMethodDetails();
+            return new AuthMethodDetails();
+        }
+    }
+
+    private void restoreADSignIn() {
+        switch (getAuthMethod()) {
```

Comment:
It seems that we could use a common method for AD/DC

Reply:
fix it.